rnote-ui/src/workspacebrowser/filerow/action/duplicate/tests/test_dup_file.rs

```diff
@@ -0,0 +1,29 @@
+use std::{path::PathBuf, fs::File};
```

Comment:
Not a huge fan of these tests. If they fail, the residuals are not cleaned up. Also, ideally filesystem tests should be contained together somewhere, not in the crate root ( [the working directory for tests](https://users.rust-lang.org/t/cargo-run-and-cargo-test-have-different-working-directory-in-a-cargo-workspace/68884). 
A possible solution is to create a `tests` dir with the [tempfile](https://docs.rs/tempfile/latest/tempfile/) crate and do all tests inside it. It is then automatically cleaned up when it is dropped.

Reply:
And the test can be all together in a module in `duplicate.rs` , no need to split these files so extensively. Check [this](https://stackoverflow.com/questions/38995892/how-to-move-tests-into-a-separate-file-for-binaries-in-rusts-cargo) for further explanation, or even the rust book chapter on unit tests

Reply:
> Not a huge fan of these tests. If they fail, the residues are not cleaned up. Also, ideally the files for filesystem tests should be contained together somewhere, not in the crate root ( [the working directory for tests](https://users.rust-lang.org/t/cargo-run-and-cargo-test-have-different-working-directory-in-a-cargo-workspace/68884).
A possible solution is to create a tests dir with the [tempfile](https://docs.rs/tempfile/latest/tempfile/) crate and do all tests inside it. It is then automatically cleaned up when it is dropped.

Hm... I'm wondering if it's worth it to have those tests. Because they are getting a bit complicated right now, since it creates a random file, then I need to path of this random file, but this could crash since the path needn't exist etc. What do you think?

Reply:
@flxzt

Reply:
I think its okay not to have them

Reply:
@flxzt alright, then it's review-ready (again xD)